push.bash

```diff
@@ -33,4 +41,23 @@ if [ -z "$BUF_COMMAND" ]; then
   fail "$NOT_INSTALLED_MESSAGE"
 fi
 
-BUF_TOKEN="${BUF_TOKEN}" "${BUF_COMMAND}" push --tag "${GITHUB_SHA}" "${BUF_INPUT}"
+if [ "${GITHUB_REF_TYPE}" != "branch" ]; then
+  echo "reference type is not branch, skipping" >&2
+  exit 0
+fi
+
+BUF_ARGS=("--tag" "${GITHUB_SHA}")
+if [ "${GITHUB_REF_NAME}" != "main" ] && [ "${ENABLE_DRAFT}" == "true" ]; then
+  # Check that --draft is supported by running "buf push --draft example --help"
+  # and checking for "unknown flag: --draft" in the output.
+  set +e
+  BUF_HELP_OUTPUT="$("${BUF_COMMAND}" push --draft example --help 2>&1)"
+  set -e
+  if [[ "${BUF_HELP_OUTPUT}" == *"unknown flag: --draft"* ]]; then
+    fail "The installed version of buf does not support setting the draft. Please use buf v1.7.0 or newer."
+  fi
```

Comment:
Why would we not use `--version` to check for >v1.7.0?

Reply:
we can, but if we drop the support for `--draft` in a later version (might be a new major version or something), it seems a less reliable / difficult check than specifically check for the flag